### ISSUE TITLE
JavaScript parser is confused by a backtick in a string

### ISSUE THREAD

**Issue by thepheer** [PrismJS/prism#2193]
**Information:**
- Prism version: latest (1.19.0)
- Plugins: none
- Environment: https://prismjs.com/test.html#language=javascript

If some string contains a backtick and there are template literals in the following code, it may invert parsing behavior: code parts are parsed as strings and strings are parsed as code.

```js
if (foo(bar)) baz(); "`"; `foo`; `bar`; if (foo(bar)) baz(); `baz`;
```

![image](https://user-images.githubusercontent.com/5144598/73198907-f576b300-4144-11ea-9cca-aeafce5325ef.png)

**Comment by RunDevelopment** [comment]
I'll close this because your issue is caused by #1492 and will be fixed by #2032.

We don't have a fix you could use right now. Sorry.

**Comment by thepheer** [comment]
Alright. I've tested #2032 on my code, it fixes the bug indeed.
